### PR TITLE
implement getting the list of docker images.

### PR DESCRIPTION
**Pull request by imrehg**
Change-type: minor

### CODE REVIEW

File: tests/lib.rs [hunk 1]
```diff
@@ -67,6 +67,72 @@ mod tests {
...
+        use narwhal::images;
+        use super::get_client;
+
+        #[test]
+        pub fn images_parse_empty() {
+            let test_str = "[]";
+            assert_eq!(images::get_images_parse(test_str).unwrap().len(), 0);
+        }
+        #[test]
+
```
Comment (CameronDiver) [inline tests/lib.rs]: We should remove this newline

File: src/images.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+use errors::*;
+use network::get;
+use serde_json;
+use types::Client;
+
+#[derive(Serialize, Deserialize, Debug)]
```
Comment (CameronDiver) [inline src/images.rs:6]: Why are we also deriving `Serialize` here?

File: src/images.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#[serde(rename_all = "PascalCase")]
+pub struct Image {
+    id: String,
```
Comment (CameronDiver) [inline src/images.rs:9]: Also these fields all need to `pub`.

File: src/images.rs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    parent_id: String,
+    repo_tags: Vec<String>,
+    repo_digests: serde_json::Value, // TODO: replace with proper type, there are some variances
+    created: i64,
```
Comment (CameronDiver) [inline src/images.rs:13]: This should be a `u64`, anything that cannnot be negative should be implemented like that.

File: src/images.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    labels: serde_json::Value, // TODO: replace with proper type, there are some variances
+    containers: i64,
+}
+
+pub fn get_images_parse(json: &str) -> Result<Vec<Image>> {
+    let images: Vec<Image> =
+        serde_json::from_str(json).chain_err(|| "Failed to deserialize images response")?;
+
+    Ok(images)
+}
```
Comment (CameronDiver) [inline src/images.rs:26]: You could probably drop the `?` and `Ok(..)` and return the result directly, after chaining the error, like:

```
fn get_images_parse(json: &str) -> Result<Vec<Image>> {
    serde_json::from_str(json).chain_err(|| "Failed to deserialize images response")
}

File: src/images.rs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+pub fn get_images(client: Client) -> Result<Vec<Image>> {
+    let response = get(client, "/images/json").chain_err(|| "Failed to get engine version")?;
```
Comment (CameronDiver) [inline src/images.rs:29]: Error string needs to be changed.

File: tests/lib.rs [hunk 7]
```diff
@@ -67,6 +67,72 @@ mod tests {
...
+                "Containers": 5
+              }
+            ]"#;
+            assert_eq!(images::get_images_parse(test_str).unwrap().len(), 2);
+        }
+
+        #[test]
+        pub fn get_images() {
+            let c = get_client();
+            let version = images::get_images(c);
```
Comment (CameronDiver) [inline tests/lib.rs:126]: `version` -> `images`

File: src/images.rs (hunk 3 above)
Comment (imrehg) [reply src/images.rs:9]: fixed in new PR

File: src/images.rs (hunk 2 above)
Comment (imrehg) [reply src/images.rs:6]: Not sure, removed.

File: src/images.rs (hunk 4 above)
Comment (imrehg) [reply src/images.rs:13]: @CameronDiver updated it, but just for created. The other ones are weird, as I've seen real results of `"SharedSize": -1,` and `"Containers": -1,` as well, so the other ones we cannot have as `u64` probably - but with Docker I cannot really be sure what's what. Their docs just say `integer`....

**Comment by imrehg** [comment]
Superseded  by #15

**Pull request closed without merging**